README.md

```diff
@@ -1,223 +1,350 @@
-    Make this primitives available also to non rust user. 
-* A set of helpers built on top of the above primitives and external bitcoin related rust crates,
-    that can be used by anyone the want to implement an Sv2 role (EG a pool that want to support Sv2
-    and that use rust)
-* An open-source implementation of an Sv2 proxy, that can be used by miners that want to use Sv2.
-* An open-source implementation of an Sv2 pool that can be used by anyone that want to operate an Sv2
-    pool.
-* A self hosted Job Negotiator + Template Provider that can be used by Sv2 or Sv1 pool that want to
-    offer job negotiation capabilities to their costumer without messing with the pool codebase (a
-    minimum amount of modification is still required)
-
-## STRUCTURE
-
-This workspace is divided in 6 macro-areas, (1) *protocols* several Stratum library, (2) *roles* the
-actual Sv2 roles, (3) *utils*, (4) *examples*, (5) *test* integration tests, (6) *experimental* not yet
-specified part of the protocol or extensions.
-
-Every dependency related to benchmarking and testing can be always opted out so we never list
-them under **External dependencies**.
-
-### PROTOCOLS
-
-Under *protocols* there is *v1* that contain an Sv1 library and *v2* that contain several Sv2
-libraries.
-
-### PROTOCOLS/V1
-TODO
-
-### PROTOCOLS/V2
-TODO
-
-### PROTOCOLS/V2/CONST-SV2
-
-A bunch of Sv2 related constants.
+* These libraries can all be used, or only a subset of them can be used, depending on the
+  functionality desired by the end user. Some examples of individuals who benefit from the
+  libraries in this repository are as follows (note that not all features are completed yet):
+    * A miner who runs a mining farm with Sv2-compatible mining firmware mining to a 
+      Sv2-compatible pool can use this library as a proxy which allows them to use the Group
+      Channels, which reduce their bandwidth consumption and bolster their efficiency. Standard
+      Channels are also supported in here, but are not as efficient as Group Channels.
+    * A miner who runs a mining farm with Sv2-compatible mining firmware mining to a Sv2-compatible
+      pool, who wants to select their own transactions to build their own `blocktemplate`, can use
+      this library as a proxy in conjunction with a Bitcoin Core node (with the Template Provider
+      logic) to do so.
+    * A miner who runs a mining farm with SV1-compatible mining firmware mining to a Sv2-compatible
+      pool, who wants to gain some of the security and efficiency improvements that Sv2 offers over
+      Stratum V1 (Sv1) (by using Extended Channels). This Sv1<->Sv2 miner proxy does not support
+      all the features of Sv2, therefore it should be used as a temporary measure before completely
+      upgrading to Sv2-compatible firmware.
+    * A pool that wants to build a Sv2 compatible pool server, complete with all Sv2 channel
```

Comment:
Here we have 3 possible users:
1. a pool that want to support Sv2: it can just deploy the open source binary crate called pool under roles, and offer an Sv2 compatible pool to their clients
2. a pool that want to build an Sv2 compatible pool in rust: will likely use the (2) rust helper library 
3. a pool that want to build an Sv2 compatible pool not in rust: will likely use the (1) rust primitives as C library

Reply:
This comment is very helpful.